### PR TITLE
Fix tests ahead of django-dynamic-fixture update

### PR DESCRIPTION
**Pull request by santos22**
Closes https://github.com/readthedocs/readthedocs.org/issues/6886

Latest pyup dependency update - https://github.com/readthedocs/readthedocs.org/pull/7012

There were a couple of tests failing locally that looked unrelated to the django-dynamic-fixure upgrade:
- ~~test_privacy_urls.py~~ (fixed)
- ~~test_core_utils.py~~ (fixed)

### CODE REVIEW

File: requirements/testing.txt [hunk 1]
```diff
@@ -1,7 +1,7 @@
 -r pip.txt
 -r local-docs-build.txt
 
-django-dynamic-fixture==2.0.0
+django-dynamic-fixture==3.1.0
```
Comment (santos22) [inline requirements/testing.txt:4]: Included this for the Travis CI build but can remove.

Comment (stsewd) [reply requirements/testing.txt:4]: Tests are still failing with the upgrade, we can let the previous version, so we can merge this and do the rest in another PR if you want.

Comment (santos22) [reply requirements/testing.txt:4]: It'd be good to fix those tests in this PR but I'm not sure where to start looking.

I've looked at the `setUp()` methods in test_privacy_urls.py and test_core_utils.py but I don't see anything wrong with the way the dynamic models are set.

Comment (santos22) [reply requirements/testing.txt:4]: Fixed the test_privacy_urls.py tests 👍

**Comment by santos22** [comment]
Looks like the following tests keep failing because `build_queue` is not set on the project:
- test_trigger_build_rounded_time_limit
- test_trigger_build_translation_medium_priority
- test_trigger_external_build_low_priority

File: requirements/testing.txt (hunk 1 above)
Comment (santos22) [reply requirements/testing.txt]: Fixed test_core_utils.py 👍

Comment (stsewd) [reply requirements/testing.txt]: We can put back the django-dynamic upgrade here.

File: readthedocs/rtd_tests/tests/test_core_utils.py [hunk 2]
```diff
@@ -166,6 +166,7 @@ def test_trigger_build_invalid_time_limit(self, update_docs):
     def test_trigger_build_rounded_time_limit(self, update_docs):
         """Time limit should round down."""
         self.project.container_time_limit = 3
+        self.project.build_queue = 'build_queue'
```
Comment (stsewd) [inline readthedocs/rtd_tests/tests/test_core_utils.py:169]: Not sure why we need this to make the tests pass. Where are we expecting this to be set?

Comment (santos22) [reply readthedocs/rtd_tests/tests/test_core_utils.py:169]: This was the last build that failed:
https://travis-ci.org/github/readthedocs/readthedocs.org/jobs/688452656

File: requirements/testing.txt (hunk 1 above)
Comment (santos22) [reply requirements/testing.txt]: Won't the upgrade be made by pyup?
https://github.com/readthedocs/readthedocs.org/pull/7012

File: readthedocs/rtd_tests/tests/test_core_utils.py (hunk 2 above)
Comment (santos22) [reply readthedocs/rtd_tests/tests/test_core_utils.py:169]: There are `AssertionError`s near the bottom.

Comment (stsewd) [reply readthedocs/rtd_tests/tests/test_core_utils.py:169]: For those cases, we just need to delete the expected value of queue `'queue': mock.ANY,`

Comment (santos22) [reply readthedocs/rtd_tests/tests/test_core_utils.py:169]: Will go ahead and remove 👍

File: requirements/testing.txt (hunk 1 above)
Comment (stsewd) [reply requirements/testing.txt]: We can do the update here, so we know all tests pass with the new version

**Comment by santos22** [comment]
Oh. I was not getting those test failures. Will fix them too...

**Pull request merged**